pwnlib/rop/rop.py

```diff
@@ -1227,6 +1239,23 @@ def __getattr__(self, k):
             leave = None
         self.leave = leave
 
+        #
+        # Load ROP.jmp_esp ('i386') or ROP.jmp_rsp ('amd64') gadgets
+        #
+        jmp_sp = {
+            4: 'jmp esp',
+            8: 'jmp rsp'
+        }[context.bytes]
```

Comment:
Don't shell out to `asm()` here, hard-code the byte values since they are known.  `as` is slow and these will never change.  I'd also prefer using `context.arch` instead of `context.bytes`.

Reply:
On second thought, they're the same for both architectures so we can just hard-code a single file.

Reply:
Using the hard-code value is really better. I'm fixing it.

I've used `context.bytes` as it is used at https://github.com/Gallopsled/pwntools/blob/dev/pwnlib/rop/rop.py#L1183-L1186. I am only pointing out, because maybe `context.arch` should be used there too.